cmake/check_amd_gpu.hip

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
What's goal for this?

Reply:
We compile & run this code during cmake build to check if an AMD GPU actually exists on the system. Only checking cuda/rocm installation is not enough as we sometimes install both during dev (for hipify-clang).